include/clap/ext/draft/configurable-audio-ports.h

```diff
@@ -30,27 +26,19 @@ typedef struct clap_audio_port_configuration_request {
    // - CLAP_PORT_MONO: (discard)
    // - CLAP_PORT_STEREO: (discard)
    // - CLAP_PORT_SURROUND: const uint8_t *channel_map
-   // - CLAP_PORT_AMBISONIC: const clap_ambisonic_info_t *info
+   // - CLAP_PORT_AMBISONIC: const clap_ambisonic_config_t *info
    const void *port_details;
 } clap_audio_port_configuration_request_t;
 
 typedef struct clap_plugin_configurable_audio_ports {
-   // Some ports may not be configurable, or simply the result of another port configuration.
-   // For example if you have a simple delay plugin, then the output port must have the exact
-   // same type as the input port; in that example, we consider the output port type to be a
-   // function (identity) of the input port type.
-   // [main-thread && !active]
-   bool(CLAP_ABI *is_port_configurable)(const clap_plugin_t *plugin,
-                                        bool                 is_input,
-                                        uint32_t             port_index);
-
-   // Submit a bunch of configuration requests which will atomically be applied together,
-   // or discarded together.
+   // If is_dry_run is true, then checks if the configuration can be applied.
+   // If is_dry_run is false, then applies the configuration.
+   // Returns true if applied.
    // [main-thread && !active]
-   bool(CLAP_ABI *request_configuration)(
-      const clap_plugin_t                                *plugin,
-      const struct clap_audio_port_configuration_request *requests,
-      uint32_t                                            request_count);
+   bool(CLAP_ABI *apply_configuration)(const clap_plugin_t                                *plugin,
```

Comment:
I guess the plugin should not call `clap_host_ambisonic::changed` or `clap_host_surround::changed` afterwards if it `apply_configuration` already returned true. Or should it? Should we document that here?

Also: should we give a hint that the host should call `clap_plugin_audio_ports::get` after successful `apply_configuration` to get information about the resulting configuration?

Reply:
Good catch, I've added some comments.